lib/pg_ha_migrations/relation.rb

```diff
@@ -77,6 +77,42 @@ def partitions(include_sub_partitions: false)
     end
   end
 
+  class TableWithLock < Table
+    def self.from_table_name(table, mode)
+      super(table).tap do |table_with_lock|
+        table_with_lock.mode = mode
+      end
+    end
+
+    attr_reader :mode
+
+    def initialize(name, schema, mode=nil)
+      super(name, schema)
+
+      self.mode = mode
+    end
+
+    def mode=(mode)
+      if mode.present?
+        @mode = LockMode.new(mode)
+      else
+        @mode = mode
+      end
+    end
+
+    def conflicts_with?(other)
+      mode.nil? || other.mode.nil? || mode.conflicts_with?(other.mode)
```

Comment:
I think this should verify that the table matches also.

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/88/commits/b9cd3000fc10b884c60b0520365cca3e87ac1d41